config/routes.php

```diff
@@ -42,12 +46,17 @@
 
     // Auth
     Route::methods([Method::GET, Method::POST], '/login')
+        ->middleware(LimitRequestsMiddleware::class)
         ->action([AuthController::class, 'login'])
         ->name('auth/login'),
     Route::post('/logout')
         ->action([AuthController::class, 'logout'])
         ->name('auth/logout'),
     Route::methods([Method::GET, Method::POST], '/signup')
+        ->middleware(fn(\Psr\Container\ContainerInterface $container) => new LimitRequestsMiddleware(
```

Comment:
You can inject only needed objects without using the container. It's just the same, but a bit cleaner